lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -762,27 +745,25 @@ private function determineIdGeneratorStrategy(AbstractPlatform $platform): int
             }
         }
 
-        foreach (self::LEGACY_DEFAULTS_FOR_ID_GENERATION as $platformFamily => $strategy) {
+        foreach (self::NON_IDENTITY_DEFAULT_STRATEGY as $platformFamily => $strategy) {
             if (is_a($platform, $platformFamily)) {
-                $recommendedStrategyName = self::RECOMMENDED_STRATEGY[$platformFamily];
-                if ($strategy !== constant('Doctrine\ORM\Mapping\ClassMetadata::GENERATOR_TYPE_' . $recommendedStrategyName)) {
+                if ($platform instanceof Platforms\PostgreSQLPlatform || is_a($platform, 'Doctrine\DBAL\Platforms\PostgreSqlPlatform')) {
```

Comment:
What is `is_a($platform, 'Doctrine\DBAL\Platforms\PostgreSqlPlatform')` checking that's not already been checked?

Reply:
There is a difference in case (`SQL` vs `Sql`). This allows supporting DBAL 2.